polars/polars-io/src/csv/write_impl.rs

```diff
@@ -265,9 +232,28 @@ pub(crate) fn write<W: Write>(
 
             let last_ptr = &col_iters[col_iters.len() - 1] as *const SeriesIter;
             let mut finished = false;
-            // loop rows
             while !finished {
+                // loop cols
                 for col in &mut col_iters {
+                    let datetime_format = match &options.datetime_format {
```

Comment:
I think it is expensive to do this branch every iteration for all dtypes. We can precompute the formats instead via a `let formats = Vec<Option<&str>` with the formats.

We can then iterate via 

```rust

for (i, col) in col_iters.enumerate() {
    let datetime_format = formats.get_unchecked(i);

...
```

Reply:
ah, right, thanks - in fact, I'd misunderstood what the code did, more fool me for not just adding some debug statements to double-check. Updating now, thanks for your review and for your patience!

Reply:
I tried this but didn't find the method `enumerate` in `col_iters`

looking into it - for now I just incremented `i`

Reply:
Does this work? `col_iters.iter().enumerate()`

Reply:
thanks, tried that but got lost in `cannot borrow `*col` as mutable, as it is behind a `&` reference` errors - I'll try again later

Reply:
ah might have it, one sec

Reply:
pushed, I was missing `iter_mut`